### PR TITLE
Ticket 10 get api articles sorting queries

### PR DESCRIPTION
**Pull request by papaparadox**
Added functionality for get request with path /api/articles that includes queries. 
Also added a test for ticket #7 , which checks if inc_votes accepts negative numbers as well.

### CODE REVIEW

File: controllers/app.controllers.js [hunk 1]
```diff
@@ -37,13 +37,24 @@ function getArticleByID(request, response, next) {
...
+      });
+  } else {
+    const { sort_by, order } = request.query;
+    selectAllArticles(sort_by, order)
+      .then((articles) => {
+        response.status(200).send({ articles: articles });
+      })
+      .catch((err) => {
+        next(err);
+      });
```
Comment (ncjaycarroll) [inline controllers/app.controllers.js:56]: I see why you've done this but it isn't necessary, if you try to destructure sort_by and order from the query and they're not there i.e. undefined then the default values you have passed to the model function will be assumed ridding the need for an if else as you have here

File: models/app.models.js [hunk 2]
```diff
@@ -17,14 +17,41 @@ function selectArticleByID(article_id) {
...
-    )
-    .then(({ rows }) => {
-      return rows;
-    });
+function selectAllArticles(sort_by = "created_at", orderToUpper = "desc") {
+  if (arguments.length === 0) {
+    return db
+      .query(
+        "SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id)::INT AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY created_at DESC"
+      )
```
Comment (ncjaycarroll) [inline models/app.models.js:25]: I'm curious, as you have passed your args default values, will the if statement on line 21 fire at all?

Comment (papaparadox) [reply models/app.models.js:25]: Yes, it runs once

File: controllers/app.controllers.js (hunk 1 above)
Comment (papaparadox) [reply controllers/app.controllers.js:56]: this catch block executes once for the scenario where rows.length === 0
